examples/getting_started/dynamic_config_and_chaining.ipynb

```diff
@@ -0,0 +1,506 @@
+     ]
+    }
+   ],
+   "execution_count": 5
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As you can see, with computed fields, you get access to both the final translation and the intermediate summary in a single response. This approach provides better traceability and can be particularly useful for debugging and understanding the entire chain of operations without the need to manage multiple separate function calls."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Advanced Techniques\n",
+    "\n",
+    "### Combining Dynamic Config and Chaining\n",
+    "\n",
+    "This example demonstrates how to use dynamic configuration within a chain of operations, all encapsulated within a single function call using computed fields."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2024-09-11T07:36:42.808355Z",
+     "start_time": "2024-09-11T07:36:23.993111Z"
+    }
+   },
+   "source": [
+    "@openai.call(\"gpt-4o-mini\")\n",
+    "@prompt_template(\"Generate a {length} {genre} story\")\n",
+    "def generate_story(genre: str, length: str): ...\n",
+    "\n",
+    "\n",
+    "@openai.call(\"gpt-4o-mini\")\n",
+    "@prompt_template(\"Analyze the tone of this story: {story}\")\n",
+    "def analyze_tone(story: str): ...\n",
+    "\n",
+    "\n",
+    "@openai.call(\"gpt-4o-mini\")\n",
+    "@prompt_template(\"Rewrite this {genre} story in a {tone} tone: {story}\")\n",
+    "def rewrite_story(story: str, genre: str, tone: str) -> openai.OpenAIDynamicConfig:\n",
+    "    return {\n",
+    "        \"computed_fields\": {\n",
+    "            \"story\": story,\n",
+    "            \"genre\": genre,\n",
+    "            \"tone\": analyze_tone(story).content,\n",
```

Comment:
nit: this should just be `analyze_tone(story)` for the improved traceability as we mentioned in the other section on chaining using computed fields